core/capabilities/ccip/ocrimpls/evm_contract_transmitter_factory.go

```diff
@@ -0,0 +1,98 @@
+package ocrimpls
+
+import (
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	ccipcommon "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/common"
+)
+
+// EVMContractTransmitterFactory implements ContractTransmitterFactory for EVM-based chains.
+type EVMContractTransmitterFactory struct{}
```

Comment:
Add the interface implementation assertion,

```golang
var _ ContractTransmitterFactory = &EVMContractTransmitterFactory{}
```

Reply:
I don't think this is needed, the type assertion happens when assigning the factory to the lookup table.